runtime/rastrace/trcengine.c

```diff
@@ -961,6 +998,31 @@ requestTraceCleanup(J9JavaVM* vm, J9VMThread * vmThread)
 	}
 }
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+/**
+ * CRIU restore initializes the trace options using an option file.
+ *
+ * @param[in] thr vmThread token
+ *
+ * @return return TRUE if success, otherwise FALSE
+ */
+static BOOLEAN
+criuRestoreInitializeTrace(J9VMThread *thr)
+{
+	J9JavaVM *vm = thr->javaVM;
+	BOOLEAN result = FALSE;
+
+	if (OMR_ERROR_NONE == enableMethodTraceHooks(vm)) {
```

Comment:
We only need to enable this if java method tracing is enabled

Reply:
Likewise with `vm->internalVMFunctions->addInternalJVMClassIterationRestoreHook(thr, setRAMClassExtendedMethodFlagsHelper);` it only needs to be called if there are methods in the `traceMethodTable`

Reply:
`traceInitializationHelper` is probably the only one that needs to be called unconditionally

Reply:
Agreed, this is enabled when `(NULL != j9ras->traceMethodTable) || (NULL != j9ras->triggerOnMethods)`.